Fix typos

kianmeng opened the pull request:
Found via `codespell -H -S ./lib/spack/external,./var/spack/repos/builtin,.git -L ser,fpr,afile,oce,doesnt,te,clen,hsa,nstall,ans,datas,onl`

haampie commented on lib/spack/llnl/util/lock.py:
```diff
@@ -68,7 +68,7 @@ class OpenFileTracker(object):
        process's lockfiles. ``fcntl`` locks are not inherited across forks, so we'll
        just track new lockfiles in the child.
-    2. Tracking by inode ensures that referencs are counted per inode, and that we don't
+    2. Tracking by inode ensures that references are counted per inode, and that we don't
        inadvertently close a file whose inode still has open locks.
```
```suggestion
    2. Tracking by inode ensures that references are counted per inode, and that we
       don't inadvertently close a file whose inode still has open locks.
```

haampie reviewed (dismissed):
Thanks!

alalazo commented:
I missed this gem @kianmeng Can you please rebase and ping me back? I'll try to give a fast track to this PR. Thanks!

alalazo commented on lib/spack/spack/build_systems/rocm.py:
```diff
@@ -69,7 +69,7 @@
 #    and introduce spec version numbers to ensure reproducible results.
 #
-# 3. In part 2, DO NOT list the path to hsa as /opt/rocm/hsa ! You want spack
+# 3. In part 2, DO NOT list the path to has as /opt/rocm/has ! You want spack
```
This seems to be a wrong fix.

alalazo reviewed (changes requested):
I found only one change to be undone.

alalazo commented:
Also:
```
lib/spack/llnl/util/lock.py:71: [E501] line too long (89 > 88 characters)
```

alalazo commented on lib/spack/spack/test/concretize.py:
```diff
@@ -1713,7 +1713,7 @@ def test_best_effort_coconcretize(self, specs, expected):
     ])
     def test_best_effort_coconcretize_preferences(
-            self, specs, expected_spec, occurances
+            self, specs, expected_spec, occurrences
```
You need to fix also the fixture name at line 1698

kianmeng replied:
@alalazo Done. Please check.

alalazo reviewed (changes requested):
I think the last bit are issues with documentation:
```
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30839/lib/spack/docs/_spack_root/lib/spack/llnl/util/filesystem.py:docstring of llnl.util.filesystem.mkdirp:: WARNING: py:obj reference target not found: permission bits
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30839/lib/spack/docs/_spack_root/lib/spack/llnl/util/filesystem.py:docstring of llnl.util.filesystem.mkdirp:: WARNING: py:obj reference target not found: group name
```
When docs build this is mergeable.

kianmeng commented:
@alalazo Sorry, I missed this notification.

When I tried to set up documentation test locally following this guide, https://spack.readthedocs.io/en/latest/contribution_guide.html#documentation-tests, I've encountered the error below:

```sh
$ git clone -c feature.manyFiles=true https://github.com/spack/spack.git
$ cd spack/bin
$ ./spack clean -ab
$ ./spack install py-sphinx py-sphinxcontrib-programoutput py-sphinx-rtd-theme graphviz git mercurial subversion
==> Bootstrapping clingo from pre-built binaries
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-dtwevigmwgke4g6ee5byktpmzmrp2kvx.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64/gcc-10.2.1/clingo-bootstrap-spack/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-dtwevigmwgke4g6ee5byktpmzmrp2kvx.spack
==> Installing "clingo-bootstrap@spack%gcc@10.2.1~docs~ipo+python+static_libstdcpp build_type=Release arch=linux-centos7-x86_64" from a buildcache
==> Bootstrapping patchelf from pre-built binaries
==> Bootstrapping patchelf@0.13.1:0.13.99 %gcc target=x86_64 from sources
==> Installing patchelf-0.13.1-dsjzmt4wboogb6lrfycmit6tdxpni7pd
==> No binary for patchelf-0.13.1-dsjzmt4wboogb6lrfycmit6tdxpni7pd found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/08/08c0237e89be74d61ddf8f6ff218439cdd62af572d568fb38913b53e222831de.tar.gz
==> No patches needed for patchelf
==> patchelf: Executing phase: 'autoreconf'
==> patchelf: Executing phase: 'configure'
==> patchelf: Executing phase: 'build'
==> patchelf: Executing phase: 'install'
==> patchelf: Successfully installed patchelf-0.13.1-dsjzmt4wboogb6lrfycmit6tdxpni7pd
  Fetch: 0.68s.  Build: 5.45s.  Total: 6.13s.
[+] /home/foobar/.spack/bootstrap/store/linux-ubuntu22.04-x86_64/gcc-11.2.0/patchelf-0.13.1-dsjzmt4wboogb6lrfycmit6tdxpni7pd
==> Error: No versions available for package 'py-sphinx'
```

Do you have any ideas on this?

alalazo commented:
@kianmeng Ugh, I lost track of this PR. I see it has a lot of conflicts now, so wondering if you're still interested in this and if you'd like to rebase / submit a new PR. Let me know.

kianmeng committed:
Fix typos

Found via `codespell -H -S ./lib/spack/external,./var/spack/repos/builtin,.git -L ser,fpr,afile,oce,doesnt,te,clen,hsa,nstall,ans,datas,onl`

kianmeng commented:
@alalazo Please check again. Thanks.

alalazo commented on lib/spack/spack/fetch_strategy.py:
```diff
@@ -152,7 +152,7 @@ def archive(self, destination):
 
     @property
-    def cachable(self):
+    def cacheable(self):
```
Let's just do comments and docstrings in this PR, and avoid functions/methods/classes

alalazo commented on lib/spack/spack/solver/asp.py:
```diff
@@ -1750,13 +1750,13 @@ def spec_clauses(self, *args, **kwargs):
         raises a comprehensible error message in case of failure.
         """
-        requestor = kwargs.pop("required_from", None)
+        requester = kwargs.pop("required_from", None)
```
This one seems ok, since the scope of the change is tiny

alalazo commented on lib/spack/spack/stage.py:
```diff
@@ -482,7 +482,7 @@ def fetch(self, mirror_only=False, err_msg=None):
                 )
 
-            if self.default_fetcher.cachable:
+            if self.default_fetcher.cacheable:
```
:point_up:

alalazo commented on share/spack/bash/spack-completion.bash:
```diff
@@ -103,7 +103,7 @@ _bash_completion_spack() {
 
     # There is no guarantee that the cursor is at the end of the command line
-    # when tab completion is envoked. For example, in the following situation:
+    # when tab completion is evoked. For example, in the following situation:
```
```suggestion
    # when tab completion is invoked. For example, in the following situation:
```

alalazo commented on share/spack/bash/spack-completion.bash:
```diff
@@ -118,7 +118,7 @@ _bash_completion_spack() {
     fi
 
-    # In general, when envoking tab completion, the user is not expecting to
+    # In general, when evoking tab completion, the user is not expecting to
```
```suggestion
    # In general, when invoking tab completion, the user is not expecting to
```

alalazo commented on share/spack/gitlab/cloud_pipelines/configs/linux/ci.yaml:
```diff
@@ -57,7 +57,7 @@ ci:
       - parallelio
       - plumed
-      - precice
+      - precise
```
Revert

alalazo commented on share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-rhel/spack.yaml:
```diff
@@ -120,7 +120,7 @@ spack:
   - petsc
   - plumed
-  - precice
+  - precise
```
Revert

alalazo commented on share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-sles/spack.yaml:
```diff
@@ -136,7 +136,7 @@ spack:
   # - petsc                       # petsc: SyntaxError: (unicode error) \N escapes not supported (can't load unicodedata module)
   # - plumed                      # python: Could not find platform dependent libraries <exec_prefix>
-  # - precice                     # petsc: SyntaxError: (unicode error) \N escapes not supported (can't load unicodedata module)
+  # - precise                     # petsc: SyntaxError: (unicode error) \N escapes not supported (can't load unicodedata module)
```
Can you revert anything that is not comment / docstring ?

kianmeng committed:
Update share/spack/bash/spack-completion.bash

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

kianmeng committed:
Update share/spack/bash/spack-completion.bash

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

kianmeng committed:
Code review fixes

alalazo commented:
@kianmeng Would you mind if we push to this PR, to solve conflicts (to avoid bothering you frequently with rebases)?

kianmeng commented:
@alalazo Go ahead. Thanks.
